pkg/image/registry/image/strategy.go

```diff
@@ -129,6 +129,38 @@ func (imageStrategy) clearSignatureDetails(image *api.Image) {
 	}
 }
 
+// internalStrategy implements behaviour for signatures subset of image.
+type internalStrategy struct {
+	imageStrategy
+}
+
+// InternalStrategy is the default logic that applies when updating
+// image' signatures via the REST API.
+var InternalStrategy = internalStrategy{Strategy}
+
+// Canonicalize normalizes the object after validation.
+func (internalStrategy) Canonicalize(obj runtime.Object) {
+}
+
+// PrepareForUpdate currently allows to set just image signatures.
+func (s internalStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	oldImage := old.(*api.Image)
+	image := obj.(*api.Image)
+
+	image.Labels = oldImage.Labels
+
+	image.DockerImageReference = oldImage.DockerImageReference
```

Comment:
He wrote a fuzz test to cover them.

On Mon, Jul 18, 2016 at 8:14 PM, Clayton Coleman <notifications@github.com>
wrote:

> In pkg/image/registry/image/strategy.go
> <https://github.com/openshift/origin/pull/9181#discussion_r71254118>:
>
> > +// InternalStrategy is the default logic that applies when updating
> > +// image' signatures via the REST API.
> > +var InternalStrategy = internalStrategy{Strategy}
> > +
> > +// Canonicalize normalizes the object after validation.
> > +func (internalStrategy) Canonicalize(obj runtime.Object) {
> > +}
> > +
> > +// PrepareForUpdate currently allows to set just image signatures.
> > +func (s internalStrategy) PrepareForUpdate(obj, old runtime.Object) {
> > +	oldImage := old.(*api.Image)
> > +	image := obj.(*api.Image)
> > +
> > +	image.Labels = oldImage.Labels
> > +
> > +	image.DockerImageReference = oldImage.DockerImageReference
>
> This is unsafe because you could unintentionally forget copying something
> new. Do it where only things explicitly known are preserved in new.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9181/files/ec22b767710a7fb1cf5496bb96ea668412df8f59#r71254118>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH2BSi2UphCDwKfloPBKFOlWLqK70b5Nks5qXBbzgaJpZM4IvEFx>
> .
>